src/moscot/base/output.py

```diff
@@ -369,3 +369,50 @@ def _ones(self, n: int) -> ArrayLike:
         import jax.numpy as jnp
 
         return jnp.ones((n,), dtype=self.transport_matrix.dtype)
+
+
+class BaseNeuralOutput(BaseSolverOutput, ABC):
```

Comment:
Consider merging `BaseNeuralOutput` and `BaseCondNeuralOutput` into one (making `condition` optional).